### PR TITLE
Communicability fix

### PR DESCRIPTION
**Pull request by Michael-E-Rose**
This commit is via #1239:

- Split `communicability_alg.py`, because it wrongly confused subgraph centrality and communicability centrality:
    - Since communicability is not a centrality, the respective part is in `networkx/algorithms`
    - The part on subgraph centrality and communicability betweenness centrality together with the Estrada index forms the new `subgraph.py` in `networkx/algorithms/centrality`
- (Hopefully) all docs updated

Note: this pull request renders #1950, #1887 and #1466 obsolete.

### CODE REVIEW

**Comment by Michael-E-Rose** [comment]
So this is finally done. Having tried with three other pull requests, but at least I learned a lot.

File: doc/source/reference/algorithms.centrality.rst [hunk 1]
```diff
@@ -57,17 +57,15 @@ Eigenvector
...
 .. autosummary::
    :toctree: generated/
 
-   communicability
-   communicability_exp
-   communicability_centrality
-   communicability_centrality_exp
-   communicability_betweenness_centrality
-   estrada_index
+    subgraph_centrality
```
Comment (jfinkels) [inline doc/source/reference/algorithms.centrality.rst]: One too many spaces in the indentation. Should be aligned with the first "a" on line 62.

File: doc/source/reference/algorithms.centrality.rst [hunk 2]
```diff
@@ -57,17 +57,15 @@ Eigenvector
...
+    subgraph_centrality_exp
+    communicability_betweenness_centrality
```
Comment (jfinkels) [inline doc/source/reference/algorithms.centrality.rst]: These should be alphabetized.

File: doc/source/reference/algorithms.centrality.rst (hunk 1 above)
Comment (jfinkels) [inline doc/source/reference/algorithms.centrality.rst]: Oh sorry, I think you *corrected* the indentation!

File: doc/source/reference/algorithms.centrality.rst [hunk 3]
```diff
@@ -57,17 +57,15 @@ Eigenvector
    katz_centrality
    katz_centrality_numpy
 
-Communicability
+Subgraph
 ---------------
```
Comment (dschult) [inline doc/source/reference/algorithms.centrality.rst]: The underline has to match the length of the word for it to format without complaining.

File: doc/source/reference/algorithms.centrality.rst (hunk 1 above)
Comment (Michael-E-Rose) [inline doc/source/reference/algorithms.centrality.rst]: *correcting* is maybe the wrong word: I removed entries that don't fit here any longer. They are now in ` doc/source/reference/algorithms.communicability.rst`.

**Comment by Michael-E-Rose** [comment]
I did the changes on `doc/source/reference/algorithms.communicability.rst` as requested by you guys. I already squashed the corresponding change into the first commit. Thanks for the hints, I wasn't aware of that.

**Pull request merged**